tests/performance/local/spinlock_overhead2.cpp

```diff
@@ -174,11 +178,20 @@ namespace test
         {
             HPX_ITT_SYNC_RELEASING(this);
 
-#if !defined( BOOST_SP_HAS_SYNC )
+#if defined(BOOST_VERSION) && BOOST_VERSION >= 107400
+#  if !defined( BOOST_SP_HAS_SYNC_INTRINSICS )
             HPX_COMPILER_FENCE;
             *const_cast<std::uint64_t volatile*>(&v_) = 0;
+#  else
+            __sync_lock_release(&v_);
+#  endif
 #else
+#  if !defined( BOOST_SP_HAS_SYNC )
+            HPX_COMPILER_FENCE;
+            *const_cast<std::uint64_t volatile*>(&v_) = 0;
+#  else
             __sync_lock_release(&v_);
+#  endif
 #endif
```

Comment:
How about this instead:
```suggestion
#if defined(BOOST_SP_HAS_SYNC_INTRINSICS) || defined(BOOST_SP_HAS_SYNC)
            __sync_lock_release(&v_);
#else
            HPX_COMPILER_FENCE;
            *const_cast<std::uint64_t volatile*>(&v_) = 0;
#endif
```
so we don't need to duplicate the two branches? The first branch can additionally be guarded by a version check if you like that.